--- luajit-51/Makefile.orig	2025-12-12 20:36:17.000000000 +0100
+++ luajit-51/Makefile	2025-12-12 22:41:05.254688084 +0100
@@ -42,7 +42,7 @@
 INSTALL_DEFINC= $(DPREFIX)/include/luajit-$(MMVERSION)
 INSTALL_INC=   $(INSTALL_DEFINC)
 
-export INSTALL_LJLIBD= $(INSTALL_SHARE_)/luajit-$(MMVERSION)
+export INSTALL_LJLIBD= $(INSTALL_SHARE_)/luajit/$(MMVERSION)
 INSTALL_JITLIB= $(DESTDIR)$(INSTALL_LJLIBD)/jit
 INSTALL_LMODD= $(INSTALL_SHARE)/lua
 INSTALL_LMOD= $(INSTALL_LMODD)/$(ABIVER)
@@ -85,7 +85,8 @@
 LDCONFIG= ldconfig -n 2>/dev/null
 SED_PC= sed -e "s|^prefix=.*|prefix=$(PREFIX)|" \
 	    -e "s|^multilib=.*|multilib=$(MULTILIB)|" \
-	    -e "s|^relver=.*|relver=$(RELVER)|"
+	    -e "s|^relver=.*|relver=$(RELVER)|" \
+	    -e "s|^version=.*|version=$(VERSION)|"
 ifneq ($(INSTALL_DEFINC),$(INSTALL_INC))
   SED_PC+= -e "s|^includedir=.*|includedir=$(INSTALL_INC)|"
 endif
--- luajit-51/src/luaconf.h.orig	2025-12-12 20:36:17.000000000 +0100
+++ luajit-51/src/luaconf.h	2025-12-12 22:36:43.720569851 +0100
@@ -50,18 +50,19 @@
 #endif
 
 #ifndef LUA_LJDIR
-#define LUA_LJDIR	LUA_JROOT "/share/luajit-2.1"
+#define LUA_LJDIR	LUA_JROOT "/share/luajit" MAJVER "." MINVER "/"
 #endif
 
 #define LUA_JPATH	";" LUA_LJDIR "/?.lua"
 #define LUA_LLDIR	LUA_LROOT "/share" LUA_LUADIR
 #define LUA_LCDIR	LUA_LROOT "/" LUA_LMULTILIB LUA_LUADIR
 #define LUA_LLPATH	";" LUA_LLDIR "?.lua;" LUA_LLDIR "?/init.lua"
+#define LUA_JCPATH	";" LUA_ROOT "/" LUA_MULTILIB LUA_LJDIR "?.so"
 #define LUA_LCPATH1	";" LUA_LCDIR "?.so"
 #define LUA_LCPATH2	";" LUA_LCDIR "loadall.so"
 
 #define LUA_PATH_DEFAULT	"./?.lua" LUA_JPATH LUA_LLPATH LUA_RLPATH
-#define LUA_CPATH_DEFAULT	"./?.so" LUA_LCPATH1 LUA_RCPATH LUA_LCPATH2
+#define LUA_CPATH_DEFAULT	"./?.so" LUA_JCPATH LUA_LCPATH1 LUA_RCPATH LUA_LCPATH2
 #endif
 
 /* Environment variable names for path overrides and initialization code. */
diff -ur LuaJIT.orig/src/Makefile LuaJIT/src/Makefile
--- LuaJIT.orig/src/Makefile	2022-04-29 09:13:18.000000000 +0200
+++ LuaJIT/src/Makefile	2022-05-01 14:26:35.049917859 +0200
@@ -283,7 +283,7 @@
   TARGET_XLIBS+= -lpthread
 endif
 
-TARGET_XCFLAGS+= $(CCOPT_$(TARGET_LJARCH))
+TARGET_XCFLAGS+= $(CCOPT_$(TARGET_LJARCH)) -DMAJVER=\"$(MAJVER)\" -DMINVER=\"$(MINVER)\"
 TARGET_ARCH+= $(patsubst %,-DLUAJIT_TARGET=LUAJIT_ARCH_%,$(TARGET_LJARCH))
 
 ifneq (,$(PREFIX))
