diff -ur LuaJIT.orig/Makefile LuaJIT/Makefile
--- LuaJIT.orig/Makefile	2022-05-01 14:24:18.622666652 +0200
+++ LuaJIT/Makefile	2022-05-01 14:40:11.423331452 +0200
@@ -35,7 +35,7 @@
 INSTALL_DEFINC= $(DPREFIX)/include/luajit-$(MMVERSION)
 INSTALL_INC=   $(INSTALL_DEFINC)
 
-INSTALL_LJLIBD= $(INSTALL_SHARE)/luajit-$(MMVERSION)
+INSTALL_LJLIBD= $(INSTALL_SHARE)/luajit/$(MMVERSION)
 INSTALL_JITLIB= $(INSTALL_LJLIBD)/jit
 INSTALL_LMODD= $(INSTALL_SHARE)/lua
 INSTALL_LMOD= $(INSTALL_LMODD)/$(ABIVER)
@@ -77,7 +77,8 @@
 LDCONFIG= ldconfig -n 2>/dev/null
 SED_PC= sed -e "s|^prefix=.*|prefix=$(PREFIX)|" \
 	    -e "s|^multilib=.*|multilib=$(MULTILIB)|" \
-	    -e "s|^relver=.*|relver=$(RELVER)|"
+	    -e "s|^relver=.*|relver=$(RELVER)|" \
+	    -e "s|^version=.*|version=$(VERSION)|"
 ifneq ($(INSTALL_DEFINC),$(INSTALL_INC))
   SED_PC+= -e "s|^includedir=.*|includedir=$(INSTALL_INC)|"
 endif
diff -ur LuaJIT.orig/src/luaconf.h LuaJIT/src/luaconf.h
--- LuaJIT.orig/src/luaconf.h	2022-04-29 09:13:18.000000000 +0200
+++ LuaJIT/src/luaconf.h	2022-05-01 14:36:10.660525525 +0200
@@ -37,7 +37,7 @@
 #endif
 #define LUA_LROOT	"/usr/local"
 #define LUA_LUADIR	"/lua/5.1/"
-#define LUA_LJDIR	"/luajit-2.1/"
+#define LUA_LJDIR	"/luajit/" MAJVER "." MINVER "/"
 
 #ifdef LUA_ROOT
 #define LUA_JROOT	LUA_ROOT
@@ -55,11 +55,12 @@
 #define LUA_LLDIR	LUA_LROOT "/share" LUA_LUADIR
 #define LUA_LCDIR	LUA_LROOT "/" LUA_LMULTILIB LUA_LUADIR
 #define LUA_LLPATH	";" LUA_LLDIR "?.lua;" LUA_LLDIR "?/init.lua"
+#define LUA_JCPATH	";" LUA_ROOT "/" LUA_MULTILIB LUA_LJDIR "?.so"
 #define LUA_LCPATH1	";" LUA_LCDIR "?.so"
 #define LUA_LCPATH2	";" LUA_LCDIR "loadall.so"
 
 #define LUA_PATH_DEFAULT	"./?.lua" LUA_JPATH LUA_LLPATH LUA_RLPATH
-#define LUA_CPATH_DEFAULT	"./?.so" LUA_LCPATH1 LUA_RCPATH LUA_LCPATH2
+#define LUA_CPATH_DEFAULT	"./?.so" LUA_JCPATH LUA_LCPATH1 LUA_RCPATH LUA_LCPATH2
 #endif
 
 /* Environment variable names for path overrides and initialization code. */
diff -ur LuaJIT.orig/src/Makefile LuaJIT/src/Makefile
--- LuaJIT.orig/src/Makefile	2022-04-29 09:13:18.000000000 +0200
+++ LuaJIT/src/Makefile	2022-05-01 14:26:35.049917859 +0200
@@ -283,7 +283,7 @@
   TARGET_XLIBS+= -lpthread
 endif
 
-TARGET_XCFLAGS+= $(CCOPT_$(TARGET_LJARCH))
+TARGET_XCFLAGS+= $(CCOPT_$(TARGET_LJARCH)) -DMAJVER=\"$(MAJVER)\" -DMINVER=\"$(MINVER)\"
 TARGET_ARCH+= $(patsubst %,-DLUAJIT_TARGET=LUAJIT_ARCH_%,$(TARGET_LJARCH))
 
 ifneq (,$(PREFIX))
